src/AsVersionTrait.php

```diff
@@ -46,7 +46,20 @@ public function asVersion($version) {
             if ($value instanceof VersionableInterface) {
                 $value = $value->asVersion($version);
             }
-            if (isset($value)) {
+            else if (is_array($value) && !empty($value)) {
```

Comment:
I've answered my own question but recording here as a reminder:
Question: Is there any advantage of `!empty(value)` vs. `$value !== array()` ?
Answer: Yes. It's possible that a property that's supposed to contain a json array or object (associative array in PHP) contains null and we need to check for that too.